### PR TITLE
[CELEBORN-642] Improve metrics and update grafana

### PR DESCRIPTION
**Pull request by onebox-li**
### What changes were proposed in this pull request?
Change in grafana 

（ALL）
add:
JVMCPUTime
LastMinuteSystemLoad
AvailableProcessors
（For Master）
add:
LostWorkers
IsActiveMaster
PartitionSize
（For Worker）
add:
PushDataFailCount -> WriteDataFailCount
ReplicateDataFailCount
ReplicateDataWriteFailCount
ReplicateDataCreateConnectionFailCount
ReplicateDataConnectionExceptionCount
ReplicateDataTimeoutCount
SortedFileSize
PushDataHandshakeFailCount
RegionStartFailCount
RegionFinishFailCount
MasterPushDataHandshakeTime
SlavePushDataHandshakeTime
MasterRegionStartTime
SlaveRegionStartTime
MasterRegionFinishTime
SlaveRegionFinishTime
PotentialConsumeSpeed
UserProduceSpeed
WorkerConsumeSpeed
DeviceOSFreeBytes
DeviceCelebornFreeBytes
push usedHeapMemory/usedDirectMemory
fetch usedHeapMemory/usedDirectMemory
replicate usedHeapMemory/usedDirectMemory
remove:
dup ReserveSlotsTime

Change dashboard layout.

Fix support for multiple labels.

Modify some metrics docs.

### Why are the changes needed?
For better use of metrics.

### Does this PR introduce _any_ user-facing change?
Below metrics change name, extract some value to the label.
DeviceOSFreeCapacity(B) -> DeviceOSFreeBytes
DeviceOSTotalCapacity(B) -> DeviceOSTotalBytes
DeviceCelebornFreeCapacity(B) -> DeviceCelebornFreeBytes
DeviceCelebornTotalCapacity(B) -> DeviceCelebornTotalBytes
push usedHeapMemory/usedDirectMemory
fetch usedHeapMemory/usedDirectMemory
replicate usedHeapMemory/usedDirectMemory

### How was this patch tested?
Cluster test.

### CODE REVIEW

**Comment by FMX** [comment]
HI, can you provide some dashboard snapshots?

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java [hunk 1]
```diff
@@ -139,22 +142,27 @@ public static PooledByteBufAllocator getPooledByteBufAllocator(
...
+              allocatorsIndex.compute(
+                  moduleName,
+                  (k, v) -> {
+                    if (v == null) {
+                      v = 0;
+                    } else {
+                      v = v + 1;
+                    }
+                    return v;
+                  });
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java:156]:
```suggestion
          int index = allocatorsIndex.compute(moduleName, (k, v) -> v == null ? 0 : v + 1);
```

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java [hunk 2]
```diff
@@ -122,14 +124,15 @@ private static PooledByteBufAllocator getSharedPooledByteBufAllocator(
...
-            _allocator, "shared-pool", conf.networkAllocatorVerboseMetric(), source);
+            _allocator, "shared-pool", conf.networkAllocatorVerboseMetric(), source, null);
       }
       return _allocator;
     }
   }
 
   public static PooledByteBufAllocator getPooledByteBufAllocator(
       TransportConf conf, AbstractSource source, boolean allowCache) {
+    Map<String, String> labels = null;
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java:135]: why define it in the outer of usage scope?

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java [hunk 3]
```diff
@@ -122,14 +124,15 @@ private static PooledByteBufAllocator getSharedPooledByteBufAllocator(
         // each core should have one arena to allocate memory
         _allocator = createPooledByteBufAllocator(true, true, conf.networkAllocatorArenas());
         new NettyMemoryMetrics(
-            _allocator, "shared-pool", conf.networkAllocatorVerboseMetric(), source);
+            _allocator, "shared-pool", conf.networkAllocatorVerboseMetric(), source, null);
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java:127]: `null` is evil, it is highly discouraged. If possible, pass an empty collection instead.

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java [hunk 4]
```diff
@@ -122,14 +124,15 @@ private static PooledByteBufAllocator getSharedPooledByteBufAllocator(
...
       return _allocator;
     }
   }
 
   public static PooledByteBufAllocator getPooledByteBufAllocator(
       TransportConf conf, AbstractSource source, boolean allowCache) {
+    Map<String, String> labels = null;
     if (conf.getCelebornConf().networkShareMemoryAllocator()) {
       return getSharedPooledByteBufAllocator(conf.getCelebornConf(), source);
     } else {
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java:138]: nit: we don't need else

```patch
- if (condition) {
-   return ...;
- } else {
-   ...
- }

+ if (condition) {
+  return ...;
+ }
+ ...
```

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java (hunk 1 above)
Comment (onebox-li) [reply common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java]: thx, updated

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java (hunk 2 above)
Comment (onebox-li) [reply common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java]: thx, updated

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java (hunk 3 above)
Comment (onebox-li) [reply common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java]: thx, updated

**Comment by onebox-li** [comment]
image
image
image
image

Panels are divided into sereval rows. The settings of the panel are similar to before, pure metrics so that users can easily do transform if needed. The screenshot is a bit ugly. For more details, you can try to import it to get the dashboard and check if there is incompatible or unreasonable. I'm not 100% sure about that. But in my environment, the json can work fine. @FMX thanks for timely reply.

**Review by pan3793 — APPROVED** [review]
Thanks, the code change LGTM, wait @FMX to check the Grafana dashboard.

**Review by pan3793 — APPROVED** [review]
Thanks, the code change LGTM, wait @FMX to check the Grafana dashboard.

**Comment by FMX** [comment]
I think the Grafana dashboard looks good to me.

**Comment by pan3793** [comment]
Thanks, merged to main/0.3

**Pull request closed without merging**